library/Neovim/Plugin/Internal.hs

```diff
@@ -43,6 +43,7 @@ getFunction (EF (_, f)) = f
 
 instance HasFunctionName (ExportedFunctionality env) where
     name = name . getDescription
+    methodName = methodName . getDescription
```

Comment:
`methodName` doesn't really fit. Since this is really only used for an internal key inside a map, it's IMHO better to call it that way: e.g. `keyInInternalFunctionMap`

However, it may even be better to create a dedicated `newtype` for the key in the function map.

Also, I think that the intention of the `HasFunctionName` type class may have been this exact purpose, I can't remember though because it has been a while. I'll take a closer look at this this weekend.

Reply:
method is what neovim calls it

Reply:
added a type `NvimMethod`, but afaict the problem is that `Request`, which as a `FunctionName` field, is used for both outgoing and incoming messages, so it's a little ugly in some places.

The method is not only used internally, it's what gets sent to nvim when registering an rpc handler. It is stored in the `remote#host#define` logic in the rpc map. So I'd say naming it `NvimMethod` is quite accurate, but please change it at your pleasure.